lib/child_process.js

```diff
@@ -490,6 +492,14 @@ ObjectDefineProperty(execFile, promisify.custom, {
   value: customPromiseExecFunction(execFile)
 });
 
+function copyProcessEnvToEnv(env, name, optionEnv) {
+  optionEnv = optionEnv || {};
```

Comment:
It's not worth it to create a new empty object for every call to the function. It'd be better to just integrate the check in the conditional, for example:

```js
if (process.env[name] &&
    (!optionEnv ||
     !ObjectPrototypeHasOwnProperty(optionEnv, name))) {
```

or breaking up the conditional:

```js
if (!process.env[name])
  return;
if (optionEnv &&
    ObjectPrototypeHasOwnProperty(optionEnv, name)) {
  return;
}
env[name] = process.env[name];
```